background.js

```diff
@@ -10,21 +21,152 @@ chrome.runtime.onMessage.addListener(async (message) => {
     }
   });
 
+// Function to check if URL should be auto-archived based on regex patterns
+async function shouldAutoArchive(url) {
+  try {
+    console.debug(`[Auto-Archive Debug] Checking URL: ${url}`);
+
+    const { enable_auto_archive, match_urls, exclude_urls } = await chrome.storage.local.get([
```

Comment:
```suggestion
    const { enable_auto_archive=false, match_urls=[], exclude_urls=[] } = await chrome.storage.local.get([
```
needs some defaults here in case chrome storage is cleared